utils/utils_test.go

```diff
@@ -94,3 +102,198 @@ func TestWriteFileIfNew(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, written, test.ShouldBeTrue)
 }
+
+func TestDownloadFile(t *testing.T) {
```

Comment:
I tried a new thing, and let AI write the bulk of this test, and tweaked a few things manually afterward. It SEEMS to be good, but I'm still hesitant about AI.